lib/active_shipping/carriers/fedex.rb

```diff
@@ -460,6 +460,7 @@ def build_location_node(xml, name, location)
           xml.StreetLines(location.address1) if location.address1
           xml.StreetLines(location.address2) if location.address2
           xml.City(location.city) if location.city
+          xml.StateOrProvinceCode(location.state)
```

Comment:
the :billing_location property in options is populated by the origin object
on the server (a new ActiveShipping::Location object, which includes the
shipper's state). That's where location.state is pulling from. By adding it
to the xml build with the StateOrProvinceCode tag, we're then passing it
through to the FedEx server as expected.

On Wed, Mar 9, 2016 at 4:41 PM, Kevin McPhillips <notifications@github.com>
wrote:

> In lib/active_shipping/carriers/fedex.rb
> <https://github.com/Shopify/active_shipping/pull/347#discussion_r55594730>
> :
>
> > @@ -460,6 +460,7 @@ def build_location_node(xml, name, location)
> >            xml.StreetLines(location.address1) if location.address1
> >            xml.StreetLines(location.address2) if location.address2
> >            xml.City(location.city) if location.city
> > +          xml.StateOrProvinceCode(location.state)
>
> I'm not positive what location.state is. Is it the code, or the name?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Shopify/active_shipping/pull/347/files#r55594730>.
>



-- 
Steve Haase
c: 240.461.6170
t: @stevehaase